mongo/estimatedcountopt/etimatedcountopt.go

```diff
@@ -0,0 +1,68 @@
+package estimatedcountopt
```

Comment:
Is there a reason you didn't just add the EstinatedDocument interface to the countopt package? I'm not sure if we need another package for this.

Reply:
Because in the method `collection.EstimatedDocumentCount ` I do the following switch 
`case estimatedcountopt.OptMaxTimeMs:`, and if I will add the EstimatedDocumentCountOption to countopt I can't differntiate them.

Reply:
The `collection.EstimatedDocumentCount` methods should work like the other collection methods, with a bundle that is unbundled. Since the only option in `estimatedcountopt` is `MaxTimeMS` I think we can just move the `EstimatedDocumentCount` interface into the `countopt` package, and then only have the `OptMaxTimeMs ` implement that interface.

I guess the important thing here to note as well is that the OptMaxTimeMS option is the same thing for both `CountDocuments` and `EstimatedDocumentCount`, so we can use one type for both.